homeassistant/components/tuya/light.py

```diff
@@ -0,0 +1,377 @@
+            hsv_v = colour_data.get("v", 0)
+            return int(self.remap(hsv_v, v_range[0], v_range[1], 0, 255))
+
+        return int(self.remap(brightness, old_range[0], old_range[1], 0, 255))
+
+    def _tuya_brightness_range(self) -> tuple[int, int]:
+        if self.dp_code_bright not in self.tuya_device.status:
+            return 0, 255
+
+        bright_value = json.loads(
+            self.tuya_device.function.get(self.dp_code_bright, {}).values
+        )
+        return bright_value.get("min", 0), bright_value.get("max", 255)
+
+    @property
+    def hs_color(self) -> tuple[float, float]:
+        """Return the hs_color of the light."""
+        colour_data = json.loads(self.tuya_device.status.get(self.dp_code_colour, 0))
+        s_range = self._tuya_hsv_s_range()
+        return colour_data.get("h", 0), self.remap(
+            colour_data.get("s", 0),
+            s_range[0],
+            s_range[1],
+            HSV_HA_SATURATION_MIN,
+            HSV_HA_SATURATION_MAX,
+        )
+
+    @property
+    def color_temp(self) -> int:
+        """Return the color_temp of the light."""
+        new_range = self._tuya_temp_range()
+        tuya_color_temp = self.tuya_device.status.get(self.dp_code_temp, 0)
+        ha_color_temp = (
+            self.max_mireds
+            - self.remap(
+                tuya_color_temp,
+                new_range[0],
+                new_range[1],
+                self.min_mireds,
+                self.max_mireds,
+            )
+            + self.min_mireds
+        )
+        return ha_color_temp
+
+    @property
+    def min_mireds(self) -> int:
+        """Return color temperature min mireds."""
+        return MIREDS_MIN
+
+    @property
+    def max_mireds(self) -> int:
+        """Return color temperature max mireds."""
+        return MIREDS_MAX
+
+    def _tuya_temp_range(self) -> tuple[int, int]:
+        temp_value = json.loads(
+            self.tuya_device.function.get(self.dp_code_temp, {}).values
+        )
+        return temp_value.get("min", 0), temp_value.get("max", 255)
+
+    def _tuya_hsv_s_range(self) -> tuple[int, int]:
+        hsv_data_range = self._tuya_hsv_function()
+        hsv_s = hsv_data_range.get("s", {"min": 0, "max": 255})
+        return hsv_s.get("min", 0), hsv_s.get("max", 255)
+
+    def _tuya_hsv_v_range(self) -> tuple[int, int]:
+        hsv_data_range = self._tuya_hsv_function()
+        hsv_v = hsv_data_range.get("v", {"min": 0, "max": 255})
+        return hsv_v.get("min", 0), hsv_v.get("max", 255)
+
+    def _tuya_hsv_function(self) -> dict[str, dict]:
+        hsv_data = json.loads(
+            self.tuya_device.function.get(self.dp_code_colour, {}).values
+        )
+        if hsv_data == {}:
+            colour_data = json.loads(
+                self.tuya_device.status.get(self.dp_code_colour, 0)
+            )
+            if (
+                self.dp_code_colour == DPCODE_COLOUR_DATA_V2
+                or colour_data.get("v") > 255
```

Comment:
`None`, which is the default value if the key is missing in the dict, doesn't compare with 255.

Reply:
fixed